xen/arch/arm/p2m.c

```diff
@@ -1098,6 +1102,34 @@ int p2m_set_entry(struct p2m_domain *p2m,
         nr -= (1 << order);
     }
 
+    if ( likely(!rc) )
```

Comment:
IOMMU mapping shouldn't be in while loop since the IOMMU might not follow the memory block the CPU can deal with.
Let's imagine that CPU MMU can work with 4K, 2M, 1G memory blocks, but the IOMMU understands 64K, 1M only, and so on. That's why I didn't IOMMU mapping into the while loop. Also I assume that IOMMU mapping should be updated only if CPU mapping passed successfully.  So, I will update the comment.